src/util.c

```diff
@@ -482,7 +477,8 @@ int tempfn_random(const char *p, char **ret) {
          *         /foo/bar/.#waldobaa2a261115984a9
          */
 
-        fn = basename(p);
+        fn = strrchr(p, '/');
+        fn = fn ? fn + 1 : p;
```

Comment:
Normally I'd say that it'd be better to provide a replacement, but since the open-coded version is shorter and we avoid the ugliness with header imports, it seems reasonable to accept this.